### PR TITLE
clippy fixes

### PR DESCRIPTION
**Pull request by matthiaskrgr**

### CODE REVIEW

File: alacritty/src/window.rs [hunk 1]
```diff
@@ -345,7 +345,7 @@ impl Window {
     }
 
     pub fn set_fullscreen(&mut self, fullscreen: bool) {
-        #[cfg(macos)]
+        #[cfg(target_os = "macos")]
         {
             if self.window().simple_fullscreen() {
```
Comment (matthiaskrgr) [inline alacritty/src/window.rs:350]: this looks like it was `macos` by mistake since all `target_os =` is used otherwise

Comment (chrisduerr) [reply alacritty/src/window.rs:350]: Turns out this is already done by winit. Would you mind removing this entirely?

We should do the same for `set_simple_fullscreen`.

Comment (matthiaskrgr) [reply alacritty/src/window.rs:350]: You mean the attribute + conditional code, right?

Comment (matthiaskrgr) [reply alacritty/src/window.rs:350]: &winit::window::Window might not have a `set_simple_fullscreen` function outside of `macos`...?

I'll push a commit and to see if CI passes and squash later if that works..

Comment (chrisduerr) [reply alacritty/src/window.rs:350]: Exactly. The dependency already makes sure that it is a no-op to call this when the other fullscreen mode is already active. So we don't need to do it ourselves.

File: alacritty/src/window.rs [hunk 2]
```diff
@@ -345,13 +345,6 @@ impl Window {
     }
 
     pub fn set_fullscreen(&mut self, fullscreen: bool) {
-        #[cfg(macos)]
-        {
-            if self.window().simple_fullscreen() {
-                return;
-            }
-        }
-
```
Comment (chrisduerr) [inline alacritty/src/window.rs:354]: The same should be done for `set_simple_fullscreen`.

Comment (matthiaskrgr) [reply alacritty/src/window.rs:354]: Oh I was looking for a `cfg(..)` inside set_simple_fullscreen :sweat_smile:  sorry....

**Pull request merged**